crowbar_framework/app/models/node_object.rb

```diff
@@ -256,6 +256,27 @@ def intended_role=(value)
     @node["crowbar_wall"]["intended_role"] = value
   end
 
+  def disks_roles=(value)
+    @node["crowbar_wall"] ||= {}
+    @node["crowbar_wall"]["disks_roles"] = value
+  end
+
+  def disks_roles
+    disks_roles = crowbar_wall["disks_roles"] || {}
+
+    # Check for current unclaimed disks, and add unclaimed disk to
+    # disks_roles hash if it is not there yet.
+    # (New disk could have been added to the node)
+    unclaimed_physical_drives.each do |d, _data|
+      unique_name = unique_device_for(d)
+      unless disks_roles[unique_name]
+        crowbar_wall["disks_roles"] ||= {}
+        crowbar_wall["disks_roles"][unique_name] = { "name" => unique_name, "role" => ""}
```

Comment:
Metrics/LineLength: Line is too long. [89/80]<br>Style/SpaceInsideHashLiteralBraces: Space inside } missing.